LSPS0/README.md

```diff
@@ -0,0 +1,606 @@
+> can individually be unreliable, leading to clients that forget
+> `id`s they issued previously, or clients or LSPs thinking that
+> the other side may have restarted when it is the network
+> between them that failed.
+> Thus, random `id`s picked from a large space are the safest
+> when the client might lose any counter state (by crashing if
+> using an in-memory counter, or loss of persistent storage on
+> hardware without storage redundancy), and are resilient
+> against reconnections when both sides remained running.
+
+Clients MAY include additional information in their `id` for
+internal tracking, as long as the total `id` has sufficient
+entropy for universal uniqueness.
+
+Clients SHOULD internally impose a reasonable timeout, on
+the scale of minutes, for receiving a response for a request,
+and SHOULD treat a timeout event as a temporary server failure,
+and forget the `id` of the timed-out request.
+
+> **Rationale** The LSP might crash between the time the client
+> makes the request to the time it could complete processing of
+> the response, and thus lost track of the request.
+
+If the LSP is unable to deliver a response to the client due
+to a disconnection, it SHOULD treat this no differently from
+successfully delivering the response but the client then does
+not "follow up" on any action after that.
+
+> **Rationale** Even if the LSP delivers the response to the
+> client, the client could then crash while processing the
+> response, which means that the LSP cannot be sure that any
+> response is received by the client anyway.
+
+### Lightning Feature Bit
+
+The [BOLT7][] specification describes the `node_announcement`
+gossip message, which includes a `features` bit field.
+The [BOLT1][] specification describes the `init` message,
+which also includes a `features` bit field.
+
+[BOLT7]: https://github.com/lightning/bolts/blob/f7dcc32694b8cd4f3a1768b904f58cb177168f29/07-routing-gossip.md
+[BOLT1]: https://github.com/lightning/bolts/blob/f7dcc32694b8cd4f3a1768b904f58cb177168f29/01-messaging.md
+
+LSPs MUST set the `features` bit numbered 729
+(`option_supports_lsps`) in both the `init` message on connection
+establishment, and in their own advertised `node_announcement`.
+Clients MUST NOT set `features` bit numbered 729 in either
+context.
+
+> **Rationale** As clients are the one who must initiate any
+> BOLT8 message ID 37913 messages before LSPs respond, they need to
+> separately discover whether a peer they connect to is
+> indeed an LSPS LSP that recognizes that message.
+> Broadcasting this as part of gossip allows a client to
+> discover new LSPs by simply downloading gossip.
+> The bit 729 was chosen randomly and has no special meaning.
+
+### LSPS Specification Support Query
+
+The client can determine if an LSP supports a particular LSPS
+specification other than LSPS0 via the `method` named
+`lsps0.listprotocols`, which has the following parameters:
+
+```JSON
+{
+  "comment": "This comment is ignored"
```

Comment:
What is the reason to include this comment here?

Reply:
See succeeding Rationale section.

Reply:
Ah, so it's an example of how to define a message with parameters? Maybe we should add a separate example then, instead of introducing an unused parameter here.

Reply:
Removed `comment` argument.  Too tired to make a separate example.